api/src/endpoints/svgThumbnail.js

```diff
@@ -0,0 +1,25 @@
+import express from 'express';
+import fs from 'fs';
+import getSvgThumbnail from 'utils/2d-map';
+
+const routes = express.Router();
+const sharp = require('sharp');
+let HPA_JSON = null;
+
+routes.get('/:svgName', async (req, res) => {
+  const { svgName } = req.params;
+  const { model } = req.query;
+
+  try {
+    const svgThumbnail = await getSvgThumbnail(svgName, model)
+    res.send(svgThumbnail);
+  } catch (e) {
+    if (e.message === '404') {
+      return res.sendStatus(404);
+    }
+
+    res.status(400).send(e.message);
```

Comment:
There is a CodeQL warning here.

Reply:
I don't see any warning message in the panel `Checks`. I guess the problem is solved now.

Reply:
This has been used at multiple places for the API endpoints. Shall we do a refactoring? But if so I guess it is better to be done in a new issue. 
This vulnerability is medium, which is caused by potential sending user input data from `e.message`.  On solution might be that we filter all alphanumeric characters from `e.message`.

Reply:
> This has been used at multiple places for the API endpoints. Shall we do a refactoring? But if so I guess it is better to be done in a new issue.

Absolutely! Marking this as resolved here.